File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02938/p02938_num2_parsed.py
MOD = 10**9 + 7

def count_pairs(L, R):
    count = 0
    for x in range(L, R+1):
        for y in range(x, R+1):
            if y % x == y ^ x:
                count += 1
    return count % MOD

L, R = map(int, input().split())
result = count_pairs(L, R)
print(result)

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02938/p02938_num2_parsed.cpp
#include <iostream>
typedef long long ll;
const ll mod = 1e9 + 7;

ll countPairs(ll L, ll R) {
    ll ans = 0;
    for (ll x = L; x <= R; x++) {
        for (ll y = x; y <= R; y += x) {
            if ((y % x) == (y ^ x)) {
                ans++;
            }
        }
    }
    return ans % mod;
}

int main() {
    ll L, R;
    std::cin >> L >> R;

    ll result = countPairs(L, R);
    std::cout << result << std::endl;

    return 0;
}